File: PerlInterviewQuestions/permutation.pl
use strict;
use warnings;
use Data::Dumper;

sub getPermuationSets {
    my $str = shift;
    my @L = split("", $str);
    my @P = ();
    for(my $i = 0; $i <= $#L; $i++) {
        for(my $j = 0; $j < $#L; $j++) {
            push(@P, join("", @L));
            my $temp = $L[$j];
            $L[$j] = $L[$j+1];
            $L[$j+1] = $temp;
        }
    }
    return @P;
}

my @permutation = getPermuationSets("abc");
$Data::Dumper::Indent = 0;
print Dumper \@permutation;

File: PythonInterviewQuestions/permutation.py
def getPermutationSets(S):
    L = list(S)
    P = []
    for i in range(len(L)):
        for j in range(len(L) - 1):
            P.append("".join(L))
            L[j], L[j + 1] = L[j + 1], L[j]

    return P


def main():
    print(getPermutationSets("abc"))


if __name__ == '__main__':
    main()
